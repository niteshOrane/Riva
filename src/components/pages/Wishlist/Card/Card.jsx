import React from 'react';
import { Link } from "react-router-dom";
import Image from '../../../common/LazyImage/Image';
import * as icons from '../../../common/Icons/Icons';
import styles from './Card.module.scss';
import { URL } from '../../../../util';

function Card({ src, name, priceWas, priceIs, remove , sku}) {
  const srcImage =
    src?.indexOf('http') > -1 ? src : `${URL.baseUrlProduct}/${src}`;
  return (
    <div className={styles.cardStyle}>
      <button
        type="button"
        title="Remove"
        onClick={remove}
        className={`${styles.closeStyle} closeBtn no-border bg-transparent position-absolute`}
      >
        <icons.Close />
      </button>
      <Link to={`/product/${sku}`}>
   
      <div className={styles.imgContainer_P} >
        <div className={styles.imgContainer}>
          <Image src={srcImage} width="100%" />
        </div>
      </div>
      </Link>
      <div className="my-12px">
        <p className={`font-size-600 ${styles.boldFont}`}>{name}</p>
      </div>
      <div className="d-flex align-items-center gap-12px">
        <s className="color-grey">was ${parseFloat(priceWas)?.toFixed(2)}</s>&nbsp; &nbsp;
        <span>Now ${parseFloat(priceIs)?.toFixed(2)}</span>
      </div>
    </div>

  );
}

export default Card;
