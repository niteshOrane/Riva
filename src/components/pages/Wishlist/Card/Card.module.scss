.cardStyle {
  position: relative;
  flex-basis: 30%;

  .closeStyle {
    right: 1px;
    top: 1px;
    z-index: 1;
  }

  .imgContainer_P {
    overflow: hidden;
    margin: -8px;
    padding: 8px;
    position: relative;
    display: flex;
    .imgContainer {
      position: relative;
      padding-bottom: 100%;
      position: relative;
      -webkit-box-flex: 1;
      flex-grow: 1;
      height: 24rem;
      img {
        position: absolute;
        inset: 0px;
        margin: auto;
        max-width: 100%;
        max-height: 100%;
        background-color: rgb(255, 255, 255);
        filter: brightness(0.975);
        object-fit: contain;
      }
    }
  }
}
.boldFont{
  font-weight: bold;
}