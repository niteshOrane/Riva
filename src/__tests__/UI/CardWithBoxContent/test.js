import React from 'react';
import renderer from 'react-test-renderer';
import {cleanup, fireEvent, render} from '@testing-library/react';
import Card from '../../../components/common/Cards/CardWithBoxContent';

afterEach(cleanup);

it('render correct value in card component', () => {
  const {getByText,getByAltText} = render(
    <Card
      item={{
        image: 'assets/images/glsGirl.png',
        title: 'EID GLAM',
        description:
          'Indulge in accessories from Gucci and beyond for a look finished to perfection',
      }}
    />
  );
  expect(getByText(/EID GLAM/i)).toBeTruthy();
  expect(getByText(/Indulge in accessories from Gucci and beyond for a look finished to perfection/i)).toBeTruthy();
  expect(getByAltText("EID GLAM").src).toContain("assets/images/glsGirl.png")
});


test('Card With img, title and description | UI', () => {
  const component = renderer.create(
    <Card
      item={{
        image: 'assets/images/glsGirl.png',
        title: 'EID GLAM',
        description:
          'Indulge in accessories from Gucci and beyond for a look finished to perfection',
      }}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});



