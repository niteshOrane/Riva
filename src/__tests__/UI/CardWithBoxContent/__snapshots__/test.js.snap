// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card With img, title and description | UI 1`] = `
<div
  className="position-relative"
>
  <div>
    <img
      alt="EID GLAM"
      src="assets/images/glsGirl.png"
      width="100%"
    />
  </div>
  <div
    className="cardBoxContent"
  >
    <h3
      className="text-center"
    >
      EID GLAM
    </h3>
    <p
      className="text-center description"
    >
      Indulge in accessories from Gucci and beyond for a look finished to perfection
    </p>
    <button
      className="bg-black color-white mx-auto  btn"
    >
      <span>
        Shop Now
      </span>
      <div
        className="before"
      >
        <span
          className="material-icons-outlined"
        >
          navigate_before
        </span>
      </div>
      <div
        className="next"
      >
        <span
          className="material-icons-outlined"
        >
          navigate_next
        </span>
      </div>
    </button>
  </div>
</div>
`;
